CMakeLists.txt

```diff
@@ -330,6 +334,7 @@ if (NOT WIN32)
 endif ()
 
 add_library(freeorioncommon
+    $<$<CXX_COMPILER_ID:msvc>:STATIC>
```

Comment:
> it already is a static library for me...?

Because you are using the MSVC project provided in the repo. @o01eg apparently tried to produce a MSVC project using the cmake scripts, and there the common library apparently is specified as dynamic.